### PR TITLE
GH-1347: DMLC Ignored Authentication Failures

### PR DESCRIPTION
**Pull request by garyrussell**
Resolves https://github.com/spring-projects/spring-amqp/issues/1347

Container starts and keeps trying to connect instead of honoring
the property.

**cherry-pick to 2.2.x**

### CODE REVIEW

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java [hunk 1]
```diff
@@ -422,6 +425,22 @@ protected void actualStart() {
 		super.doStart();
 		final String[] queueNames = getQueueNames();
 		checkMissingQueues(queueNames);
+		if (isPossibleAuthenticationFailureFatal()) {
+			Connection connection = null;
+			try {
+				getConnectionFactory().createConnection();
+			}
+			catch (AmqpAuthenticationException ex) {
```
Comment (artembilan) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java:433]: In docs you say `PossibleAuthenticationFailureException` and I see that exactly this `PossibleAuthenticationFailureException` is caught in the `SimpleMessageListenerContainer.AsyncMessageProcessingConsumer`.

Shouldn't the same one be caught here as well?

**Review by artembilan — CHANGES_REQUESTED** [review]

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java (hunk 1 above)
Comment (garyrussell) [reply spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java:433]: I think that is dead code; I just ran a test and it is handled by this

https://github.com/spring-projects/spring-amqp/blob/8e8dda721f9f7a0bd2a2373ccf4c40103f3ffbdd/spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/BlockingQueueConsumer.java#L601-L603

and

https://github.com/spring-projects/spring-amqp/blob/8e8dda721f9f7a0bd2a2373ccf4c40103f3ffbdd/spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainer.java#L1361-L1364

I'll create a new issue to clean that up.

Comment (artembilan) [reply spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java:433]: OK. Should we merge then this as is?
I see that `RabbitExceptionTranslator` does the proper conversion:
```
if (ex instanceof PossibleAuthenticationFailureException) {
	return new AmqpAuthenticationException(ex);
}
```

Comment (garyrussell) [reply spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java:433]: I would say yes, if you don't see any problems with it.

**Comment by artembilan** [comment]
... and cherry-picked to `2.2.x` after fixing conflicts.

**Pull request merged**